### PR TITLE
UI redesign part 4

### PR DESCRIPTION
**Pull request by mvaivre**

### CODE REVIEW

File: apps/mobile-wallet/src/components/RoundedCard.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  style?: StyleProp<ViewStyle>
+}
+
+const RoundedCard = ({ children, style }: RoundedCardProps) => (
+  <RoundedCardContainer style={style}>{children}</RoundedCardContainer>
+)
+
+export default RoundedCard
+
+const RoundedCardContainer = styled.View`
```
Comment (nop33) [inline apps/mobile-wallet/src/components/RoundedCard.tsx:33]: The pattern we've used so far is `RoundedCardStyled` for this sort of components

File: apps/mobile-wallet/src/components/footers/FooterMenu.tsx [hunk 2]
```diff
@@ -30,13 +30,14 @@ interface FooterMenuProps extends BottomTabBarProps {
   style?: StyleProp<ViewStyle>
 }
 
+const isIos = Platform.OS === 'ios'
```
Comment (nop33) [inline apps/mobile-wallet/src/components/footers/FooterMenu.tsx:33]: nit: we have this in other places too. Let's extract it to a util somewhere or just use `Platform.OS === 'ios'` which is short enough imo

File: apps/mobile-wallet/src/components/inputs/Input.tsx [hunk 3]
```diff
@@ -81,6 +90,15 @@ const Input = <T extends InputValue>({
...
   }))
 
+  useEffect(() => {
+    const fetchCopiedText = async () => {
+      const text = await getStringAsync()
+      setCopiedText(text)
+    }
+
+    fetchCopiedText()
+  })
```
Comment (nop33) [inline apps/mobile-wallet/src/components/inputs/Input.tsx:100]: The dependency array is missing.

Also, I started preferring this pattern, WDYT?

```suggestion
useEffect(() => {
    getStringAsync().then(setCopiedText)
  }, [])
```

File: apps/mobile-wallet/src/components/inputs/Input.tsx [hunk 4]
```diff
@@ -97,6 +115,10 @@ const Input = <T extends InputValue>({
     onBlur && onBlur(e)
   }
```
Comment (nop33) [inline apps/mobile-wallet/src/components/inputs/Input.tsx:116]: Missing dependency array

File: apps/mobile-wallet/src/contexts/walletConnect/WalletConnectSessionRequestModal.tsx [hunk 5]
```diff
@@ -55,7 +61,8 @@ import { SessionRequestData } from '~/types/walletConnect'
 import { showExceptionToast } from '~/utils/layout'
 import { getTransactionAssetAmounts } from '~/utils/transactions'
 
-interface WalletConnectSessionRequestModalProps<T extends SessionRequestData> extends ModalContentProps {
+interface WalletConnectSessionRequestModalProps<T extends SessionRequestData> {
+  walletConnectClient: SignClient
```
Comment (nop33) [inline apps/mobile-wallet/src/contexts/walletConnect/WalletConnectSessionRequestModal.tsx:65]: Isn't this accessible via `useWalletConnectContext`?

File: apps/mobile-wallet/src/contexts/walletConnect/WalletConnectContext.tsx
Comment (nop33) [inline apps/mobile-wallet/src/contexts/walletConnect/WalletConnectContext.tsx:1]: what a diff...

Have you tested all possible WC methods by using the example dapp?

File: apps/mobile-wallet/src/features/modals/BottomModalHeader.tsx [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+
+import Animated from 'react-native-reanimated'
+import styled from 'styled-components/native'
+
+import AppText from '~/components/AppText'
+import { CloseButton } from '~/components/buttons/Button'
+import { DEFAULT_MARGIN } from '~/style/globalStyle'
+
+interface BottomModalHeaderProps {
+  handleClose: () => void
```
Comment (nop33) [inline apps/mobile-wallet/src/features/modals/BottomModalHeader.tsx:27]: we tend to call these props `on...` not `handle...`

**Review by nop33 — COMMENTED** [review]
Just a quick pass over the diff!

I did not do any functional testing since it's still in draft mode.

File: apps/mobile-wallet/src/contexts/walletConnect/WalletConnectContext.tsx
Comment (mvaivre) [reply apps/mobile-wallet/src/contexts/walletConnect/WalletConnectContext.tsx:1]: It's just code blocks moving around. No code changes.

File: apps/mobile-wallet/src/components/inputs/Input.tsx (hunk 3 above)
Comment (mvaivre) [reply apps/mobile-wallet/src/components/inputs/Input.tsx:100]: I also like it better I think.

File: apps/mobile-wallet/src/components/RoundedCard.tsx (hunk 1 above)
Comment (mvaivre) [reply apps/mobile-wallet/src/components/RoundedCard.tsx:33]: Done

File: apps/mobile-wallet/src/components/footers/FooterMenu.tsx (hunk 2 above)
Comment (mvaivre) [reply apps/mobile-wallet/src/components/footers/FooterMenu.tsx:33]: Done

File: apps/mobile-wallet/src/components/inputs/Input.tsx (hunk 4 above)
Comment (mvaivre) [reply apps/mobile-wallet/src/components/inputs/Input.tsx]: Code has been removed

File: apps/mobile-wallet/src/components/inputs/Input.tsx (hunk 3 above)
Comment (mvaivre) [reply apps/mobile-wallet/src/components/inputs/Input.tsx]: Changed to the pattern you proposed

File: apps/mobile-wallet/src/features/modals/BottomModalHeader.tsx (hunk 6 above)
Comment (mvaivre) [reply apps/mobile-wallet/src/features/modals/BottomModalHeader.tsx]: Done

File: apps/mobile-wallet/src/contexts/walletConnect/WalletConnectSessionRequestModal.tsx (hunk 5 above)
Comment (mvaivre) [reply apps/mobile-wallet/src/contexts/walletConnect/WalletConnectSessionRequestModal.tsx]: Udpated

**Comment by nop33** [comment]
@mvaivre before I review this PR, can you please update the base branch and rebase if necessary?

Also, we shouldn't be getting errors in the static analysis GH action anymore. At least `next` is clean.

**Pull request closed without merging**